Cauldron/Assets/Scripts/SO/MainSettings.cs

```diff
@@ -57,6 +58,10 @@ public class StatusBars
             public int ThresholdDecrement => (int)(thresholdDecrement/100 * Total);
             public int GetMinThreshold => (int)(minThreshold/100 * Total);
             public int GetMaxThreshold => (int)((100f - minThreshold)/100 * Total);
+        
+            
+            public event Func<int,int> TryGetCovenCost;
```

Comment:
Well, this looks strange to me)) We actually need a simple int here for override, you can use int? to check it for null.

Using Func looks like over-engineering in our case and I suppose you don't need an event here (or is it used to support several overrides?)

But I'm okey with it)

Reply:
I think I used int? here. Not sure if we need another override ("You aren’t gonna need it" principle, yes?).
And... We can't make a struct for settings. "The initializers of the struct fields are not available" to us in the current version of the language - that's what the compiler said...